comps/llms/summarization/predictionguard/llm_predictionguard.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (C) 2024 Prediction Guard, Inc.
+# SPDX-License-Identified: Apache-2.0
+import json
+import time
+
+from fastapi import FastAPI, HTTPException
+from fastapi.responses import StreamingResponse
+from predictionguard import PredictionGuard
+
+from comps import (
+    GeneratedDoc,
+    LLMParamsDoc,
+    ServiceType,
+    opea_microservices,
+    register_microservice,
+    register_statistics,
+    statistics_dict,
+)
+
+client = PredictionGuard()
+app = FastAPI()
+
+
+@register_microservice(
+    name="opea_service@llm_predictionguard_docsum",
+    service_type=ServiceType.LLM,
+    endpoint="/v1/chat/docsum",
+    host="0.0.0.0",
+    port=9000,
+)
+@register_statistics(names=["opea_service@llm_predictionguard_docsum"])
+def llm_generate(input: LLMParamsDoc):
+    start = time.time()
+
+    messages = [
+        {
+            "role": "system",
+            "content": "You are a summarization assistant. Your goal is to provide a very concise, summarized responses of the user query.",
```

Comment:
@lvliang-intel pls review this PR. 

From my view, why not merge this with guardrail component we have had? looks like only the prompt and model variant are different

Reply:
@ftian1 Hey Tian,

Hope you're having a good day.

> **TL;DR:** I want to add a document summarization component (PredictionGuard DocSum) to their GenAI Example Repo. This will also allow flexibility in how doc summaries are generated without altering the existing text generation component.


Adding this summarization component, is desired, because it relies on the [**PredictionGuard** ](https://docs.predictionguard.com/api-reference/api-reference/completions) platform.

I want to add a PredictionGuard DocSum Permutation to the  GenAI Example Repo at some point, and the document summarization component is a necessary component for that _MegaService_.


You are correct, that this component only differs in the prompt and model from the predictionguard/text-gen component.

Having two separate components allows us to separate concerns if in the future we want to change how docsum is handled by predictionguard, without modifying textgen